File: Unidad 1/T1.1 SIN BD/TI1.1 SOAP_JAVA_CONUNI_GR01/02 CLIMOV/02 CLIMOV/app/src/main/java/com/example/a02climov/ec/edu/monster/view/ConversionActivity.java
package ec.edu.monster.view;

import android.os.Bundle;
import android.os.Handler;
import android.os.Looper;
import android.widget.ArrayAdapter;
import android.widget.Button;
import android.widget.EditText;
import android.widget.Spinner;
import android.widget.TextView;
import android.widget.Toast;

import androidx.appcompat.app.AppCompatActivity;

import java.util.concurrent.Executors;

import ec.edu.monster.controller.ConversionController;
import ec.edu.monster.model.ConversionModel;
import com.example.a02climov.R;

public class ConversionActivity extends AppCompatActivity {

    private static final String SERVICE_URL =
            "http://192.168.100.170:8080/WS_CONUNI_SOAPJAVA_GRO01/WS_Conversion_Unidades";

    private final String[] opciones = new String[]{
            "Kilogramos a Libras",
            "Libras a Kilogramos",
            "Gramos a Onzas",
            "Celsius a Fahrenheit",
            "Celsius a Kelvin",
            "Fahrenheit a Celsius",
            "Kilómetros a Millas",
            "Metros a Pies",
            "Pulgadas a Centímetros"
    };

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_conversion);

        Spinner spTipo = findViewById(R.id.spTipo);
        EditText etValor = findViewById(R.id.etValor);
        Button btnConvertir = findViewById(R.id.btnConvertir);
        TextView tvResultado = findViewById(R.id.tvResultado);

        spTipo.setAdapter(new ArrayAdapter<>(this,
                android.R.layout.simple_spinner_dropdown_item, opciones));

        btnConvertir.setOnClickListener(v -> {
            String tipo = (String) spTipo.getSelectedItem();
            String valorStr = etValor.getText().toString().trim();
            if (valorStr.isEmpty()) {
                Toast.makeText(this, "Ingrese un valor", Toast.LENGTH_SHORT).show();
                return;
            }
            double valor = Double.parseDouble(valorStr);

            Executors.newSingleThreadExecutor().execute(() -> {
                try {
                    ConversionController ctrl = new ConversionController(SERVICE_URL);
                    ConversionModel model = new ConversionModel(valor, tipo);
                    double out = ctrl.convertir(model);

                    new Handler(Looper.getMainLooper()).post(
                            () -> tvResultado.setText("Resultado: " + out)
                    );
                } catch (Exception e) {
                    new Handler(Looper.getMainLooper()).post(
                            () -> Toast.makeText(this, "Error: " + e.getMessage(), Toast.LENGTH_LONG).show()
                    );
                }
            });
        });
    }
}

File: Unidad 1/T1.1 SIN BD/TI1.1 SOAP_DOTNET_CONUNI_GR01/02 CLIMOV/app/src/main/java/com/example/a02climov/controller/ConversionActivity.kt
package com.example.a02climov.controller

import android.os.Bundle
import android.view.View
import android.widget.*
import androidx.appcompat.app.AppCompatActivity
import androidx.lifecycle.lifecycleScope
import com.example.a02climov.R
import com.example.a02climov.model.SoapClient
import kotlinx.coroutines.launch

class ConversionActivity : AppCompatActivity() {

    private val operations = listOf(
        Triple("Kilómetros a Millas", "KilometrosAMillas", "km"),
        Triple("Metros a Pies", "MetrosAPies", "m"),
        Triple("Pulgadas a Centímetros", "PulgadasACentimetros", "p"),
        Triple("Kilogramos a Libras", "KilogramosALibras", "kg"),
        Triple("Gramos a Onzas", "GramosAOnzas", "g"),
        Triple("Libras a Kilogramos", "LibrasAKilogramos", "lb"),
        Triple("Celsius a Fahrenheit", "CelsiusAFahrenheit", "c"),
        Triple("Fahrenheit a Celsius", "FahrenheitACelsius", "f"),
        Triple("Celsius a Kelvin", "CelsiusAKelvin", "c")
    )

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_conversion)

        val spOperation = findViewById<Spinner>(R.id.spOperation)
        val etBaseUrl = findViewById<EditText>(R.id.etBaseUrl)
        val etValue = findViewById<EditText>(R.id.etValue)
        val btnConvert = findViewById<Button>(R.id.btnConvert)
        val tvResult = findViewById<TextView>(R.id.tvResult)
        val progress = findViewById<ProgressBar>(R.id.progress)

        spOperation.adapter = ArrayAdapter(
            this,
            android.R.layout.simple_spinner_dropdown_item,
            operations.map { it.first }
        )

        etBaseUrl.setText("http://10.0.2.2:57003/ConversionService.svc")

        btnConvert.setOnClickListener {
            val baseUrl = etBaseUrl.text.toString().trim()
            val value = etValue.text.toString().toDoubleOrNull()
            if (baseUrl.isEmpty()) { toast("Ingresa la URL del .svc"); return@setOnClickListener }
            if (value == null) { toast("Ingresa un valor numérico válido"); return@setOnClickListener }

            val (label, method, paramName) = operations[spOperation.selectedItemPosition]

            progress.visibility = View.VISIBLE
            btnConvert.isEnabled = false

            lifecycleScope.launch {
                try {
                    val result = SoapClient.callDoubleMethodKsoap(
                        baseUrl = baseUrl,
                        method = method,
                        param = paramName,
                        value = value
                    )
                    tvResult.text = "Resultado de $label: $result"
                } catch (e: Exception) {
                    tvResult.text = "Error"
                    toast("Error: ${e.message?.take(300)}")
                } finally {
                    progress.visibility = View.GONE
                    btnConvert.isEnabled = true
                }
            }
        }
    }

    private fun toast(msg: String) =
        Toast.makeText(this, msg, Toast.LENGTH_SHORT).show()
}
